flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointRequestDecider.java

```diff
@@ -146,15 +151,11 @@
 
 	private Optional<CheckpointTriggerRequest> onTooEarly(long nextTriggerDelayMillis) {
 		CheckpointTriggerRequest first = queuedRequests.first();
-		if (first.isForce()) {
-			return Optional.of(queuedRequests.pollFirst());
-		} else if (first.isPeriodic) {
+		if (first.isPeriodic) {
```

Comment:
Can we inline this function now?
Otherwise, we need to duplicate the check and return logic (or use preconditions).

Currently, I think `if` is missing `&& !isForce` check,
and in the end, it should return `pollFirst`.

Reply:
Yes, I will remove all `if` from this function.  The conditions-returnValue map of function `chooseRequestToExecute` would be like this:

|  isForce  | isPeriodic | too eraly | return value of `chooseRequestToExecute` | 
|  ----  | ----  | ---- | ---- |
| true  | true | true | pollFirst |
| true  | true | false | pollFirst |
| true  | false | true | pollFirst | 
| true  | false | false | pollFirst | 
| false  | true | true  | empty,  and call `queuedRequests.pollFirst().completeExceptionally` | 
| false  | true | false  | pollFirst | 
| false  | false | true  | pollFirst | 
| false  | false | false | pollFirst|

Reply:
Thanks, the table explains very clearly what `chooseRequestToExecute` will return.


> I will remove all if from this function.

WDYT about removing the function (`onTooEarly`) at all? (and moving the two remaining statements into `chooseRequestToExecute`)

Reply:
I think it's a good idea.  It will make the logic of `chooseRequestToExecute` more clearer.